puppetboard/utils.py

```diff
@@ -31,27 +32,40 @@ def jsonprint(value):
     return json.dumps(value, indent=2, separators=(',', ': '))
 
 
-def get_db_version(puppetdb):
+def check_db_version(puppetdb):
     """
-    Get the version of puppetdb.  Version form 3.2 query
-    interface is slightly different on mbeans
+    Gets the version of puppetdb and exits if it is not an accepted one.
     """
-    ver = ()
     try:
         version = puppetdb.current_version()
-        (major, minor, build) = [int(x) for x in version.split('.')]
-        ver = (major, minor, build)
-        log.info("PuppetDB Version %d.%d.%d" % (major, minor, build))
-    except ValueError:
-        log.error("Unable to determine version from string: '%s'" % puppetdb.current_version())
-        ver = (4, 2, 0)
+
+        log.info(f"PuppetDB version: {version}")
+
+        # Puppet Server is enforcing new metrics API (v2)
+        # starting with versions 6.9.1, 5.3.12, and 5.2.13
+        if parse('5.2.0') <= parse(version) < parse('5.2.13'):
```

Comment:
Nit: This is calling `parse(version)` quite a few times. Perhaps it should be stored in a variable?

Reply:
Thanks @ekohl 🙏, corrected. 😊